src/member.ts

```diff
@@ -9,13 +11,39 @@ interface MemberState {
   reputation: number
   tokens: number
   gen: number
+  approvedGen: number
```

Comment:
that is very nice (even if it costs more gas, which is not nice). But we might need this anyway because (a) we may get to do the frontend upgrade (remove the approve stuff and use approveAndCall) after doing the get-the-data-from-subnode upgrade, and I would not want to get stuck on something stupid like this and (b) even aftwr we do the  approveandcall upgrade, we may want to tell the user that _did_ approve gen that he has done so, and how much.

Reply:
ok